python/mxnet/ndarray/ndarray.py

```diff
@@ -306,8 +306,14 @@ def __le__(self, other):
         return lesser_equal(self, other)
 
     def __bool__(self):
-        raise ValueError("The truth value of an NDArray is ambiguous. " \
-                         "Please convert to number with asscalar() first.")
+        num_elements = reduce(operator.mul, self.shape, 1)
+        if num_elements == 0:
```

Comment:
Since this is for checking whether a one-element ndarray is 0 or not, is returning `False` for ndarray with incomplete shape information well-defined for that purpose, or raising an exception more appropriate?

Reply:
This is to be consistent with numpy array.
```python
In [1]: import numpy as np

In [2]: bool(np.array([]))
Out[2]: False
```

Reply:
Well the meaning of `num_elements=0` is different in MXNet and Numpy.
In Numpy, it simply means that the array has no elements, while in MXNet, it could mean one of the several following things:
1. the array is empty, and shape is (0,...)
2. the array's shape is completely unknown and needs to be inferred from others in the network
3. the array's shape is partially unknown and needs to be inferred from others in the network

Does it makes sense to return `False` for items 2 and 3?